File: ForceConnector/ForceRibbon.vb
Imports Microsoft.Office.Tools.Ribbon

Public Class ForceRibbon
    Private Sub ForceRibbon_Load(ByVal sender As System.Object, ByVal e As RibbonUIEventArgs) Handles MyBase.Load

    End Sub


    ''' <summary>
    '''  Data Connector Module
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub btnAbout_Click(sender As Object, e As RibbonControlEventArgs) Handles btnAbout.Click
        OpenAbout()
    End Sub

    Private Sub TableWizard_Click(sender As Object, e As RibbonControlEventArgs) Handles TableWizard.Click
        QueryTableWizard()
    End Sub

    Private Sub UpdateCells_Click(sender As Object, e As RibbonControlEventArgs) Handles UpdateCells.Click
        UpdateSelectedCells()
    End Sub

    Private Sub InsertRows_Click(sender As Object, e As RibbonControlEventArgs) Handles InsertRows.Click
        InsertSelectedRows()
    End Sub

    Private Sub QueryRows_Click(sender As Object, e As RibbonControlEventArgs) Handles QueryRows.Click
        QuerySelectedRows()
    End Sub

    Private Sub DescribeSobject_Click(sender As Object, e As RibbonControlEventArgs) Handles DescribeSobject.Click
        DescribeSforceObject()
    End Sub

    Private Sub QueryTable_Click(sender As Object, e As RibbonControlEventArgs) Handles QueryTable.Click
        QueryTableData()
    End Sub

    Private Sub DeleteRecords_Click(sender As Object, e As RibbonControlEventArgs) Handles DeleteRecords.Click
        DeleteSelectedRecords()
    End Sub

    Private Sub Options_Click(sender As Object, e As RibbonControlEventArgs) Handles Options.Click
        OptionsForm()
    End Sub

    Private Sub Logout_Click(sender As Object, e As RibbonControlEventArgs) Handles Logout.Click
        LogoutFrom() 'sfLogout
    End Sub


    ''' <summary>
    '''  Translation Helper Module
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>

    Private Sub btnDownloadCustomLabel_Click(sender As Object, e As RibbonControlEventArgs) Handles btnDownloadCustomLabel.Click
        METAAPI.DownloadCustomLabels()
    End Sub

    Private Sub btnDownloadObjectTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnDownloadObjectTranslation.Click
        METAAPI.DownloadObjectTranslations()
    End Sub

    Private Sub btnDownloadTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnDownloadTranslation.Click
        METAAPI.DownloadTranslations()
    End Sub

    Private Sub btnUploadCustomLabel_Click(sender As Object, e As RibbonControlEventArgs) Handles btnUploadCustomLabel.Click
        METAAPI.UploadCustomLabels()
    End Sub

    Private Sub btnUpdateObjectTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnUpdateObjectTranslation.Click
        METAAPI.UpdateObjectTranslations()
    End Sub

    Private Sub btnUpdateTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnUpdateTranslation.Click
        METAAPI.UpdateTranslations()
    End Sub

    Private Sub btnDownloadCustomLabelTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnDownloadCustomLabelTranslation.Click
        METAAPI.DownloadCustomLabelTranslations()
    End Sub

    Private Sub btnUpdateCustomLabelTranslation_Click(sender As Object, e As RibbonControlEventArgs) Handles btnUpdateCustomLabelTranslation.Click
        METAAPI.UpdateCustomLabelTranslations()
    End Sub

End Class

File: ForceConnector/ForceRibbon.cs
using Microsoft.Office.Tools.Ribbon;

namespace ForceConnector
{
    public partial class ForceRibbon
    {
        private void ForceRibbon_Load(object sender, RibbonUIEventArgs e)
        {
        }


        /// <summary>
    /// Data Connector Module
    /// </summary>
    /// <param name="sender"></param>
    /// <param name="e"></param>
        private void btnAbout_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.OpenAbout();
        }

        private void TableWizard_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.QueryTableWizard();
        }

        private void UpdateCells_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.UpdateSelectedCells();
        }

        private void InsertRows_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.InsertSelectedRows();
        }

        private void QueryRows_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.QuerySelectedRows();
        }

        private void DescribeSobject_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.DescribeSforceObject();
        }

        private void QueryTable_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.QueryTableData();
        }

        private void DeleteRecords_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.DeleteSelectedRecords();
        }

        private void Options_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.OptionsForm();
        }

        private void Logout_Click(object sender, RibbonControlEventArgs e)
        {
            ForceConnector.LogoutFrom(); // sfLogout
        }


        /// <summary>
    /// Translation Helper Module
    /// </summary>
    /// <param name="sender"></param>
    /// <param name="e"></param>

        private void btnDownloadCustomLabel_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.DownloadCustomLabels();
        }

        private void btnDownloadObjectTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.DownloadObjectTranslations();
        }

        private void btnDownloadTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.DownloadTranslations();
        }

        private void btnUploadCustomLabel_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.UploadCustomLabels();
        }

        private void btnUpdateObjectTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.UpdateObjectTranslations();
        }

        private void btnUpdateTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.UpdateTranslations();
        }

        private void btnDownloadCustomLabelTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.DownloadCustomLabelTranslations();
        }

        private void btnUpdateCustomLabelTranslation_Click(object sender, RibbonControlEventArgs e)
        {
            METAAPI.UpdateCustomLabelTranslations();
        }
    }
}